### ISSUE TITLE
pentatonic mysic research: i am a musician.

### ISSUE THREAD

**Issue by terraxoxo** [tensorflow/magenta#402]
for pentatonic music-

hello, @douglaseck    @kastnerkyle and everyone.

my name is terra, i am so glad to be here and communicate with you all. i am a professional musician who are really interested in science and technology.

i am an Asian classcial musician, actually asian traditional music is composed of pentatonic music. of course it existed in the western music but asian music was evolved in different dimension. does google have no plans and no interested in pentatonic music...? 

#401

**Comment by cghawthorne** [comment]
Yes, we're definitely interested in pentatonic music. Our existing models should be able to work with pentatonic music, though it's hard to know for sure without trying. Do you have a corpus of pentatonic music in either MIDI or MusicXML format you could use to train a model? It would be very interesting to see how well it does.

**Comment by terraxoxo** [comment]
thank you @cghawthorne.
MIDI is computer music. I don't figure out what kind of MIDI music you mean. 
I can give you some music sample:[https://soundcloud.com/poly-music-co](url)
and Music XML is new to me. In fact, almost all professional performers have no interest in these technology things. Usually composers might have some interest, though,.
I can deal with Finale or Sibelius program and if it needed, I will learn the MusicXML. [https://en.wikipedia.org/wiki/MusicXML](url)

I am assuming, the Asian music will be really excited with new technology, and it must challengeable to do something with. One of reasons is the music cannot be written for their real substance, and many of institute as like IRCAM in France had given up to digitalize for some music. 

I attach an Asian notation sample: [http://www.rutadeseda.org/corea/musica/notacion8.html](url)

thanks!

**Comment by cghawthorne** [comment]
Yeah, it sounds like the challenge will be finding a large enough dataset of symbolic data in a format that Magenta can understand. The training will work best if you have on the order of hundreds or thousands of songs to train from.

**Comment by terraxoxo** [comment]
thank you. yes I already recorded hundreds of music recording. And there are also live recordings in the famous hall as like carnegie hall. As well as, singing too.
If magenta can overcome these kinds of music, the magenta will be able to play any kinds of music on the planet. I am sure.

**Comment by DefinitlyEvil** [comment]
@terraxoxo I am in a music college majoring music composing haha, but I am a self-taught programmer as well. You can do anything with this library I think. You could throw them midi files and they can imitate the exact type of music, as for asian music, just throw them asian music midi files.

**Comment by terraxoxo** [comment]
Hi,,, Evill?
Thank you for message...but I don't understand exactly what you are
saying..? can you give me more informations..? thank you very much. ^^ and
can you let me know your name..? terra, xoxo.

**Comment by DefinitlyEvil** [comment]
@terraxoxo Sorry I can't give my name(privacy ;) ) but I can explain more. See, the machine (neural network) is completely blank and don't know anything about music. When you throw them with some midi files within specific era, it will learn from patterns from it but no one can explain how it works because everything are just weights. So it might or might not be able to learn the harmonies hidden inside a piece. I was planning to train it with Bach's French suite, but I ran into errors.

**Comment by DefinitlyEvil** [comment]
@terraxoxo Do u have Skype or something?

**Comment by avinash040** [comment]
@terraxoxo You could probably create a MIDI dataset by converting your existing music files into MIDI. There are software tools that let you do this. However, they are not very accurate. Ableton lets you convert live audio into MIDI (doesn't really work well when harmony is involved)

**Comment by DefinitlyEvil** [comment]
@avinash040 midi is based on twelve tone but for example I heard that Indian music even use notes between two semitones. xD

**Comment by terraxoxo** [comment]
@DefinitlyEvil 
Yes, I have skype, haha. I will let you know by email. but hope you can do self-introduce to me....!

**Comment by terraxoxo** [comment]
here contact me,
skype id: terra.raesook.han

**Comment by terraxoxo** [comment]
yes, I know the indian music also. I play also sitar.

**Comment by DefinitlyEvil** [comment]
@terraxoxo okay ;p cool

**Comment by Zenanimator** [comment]
**@terraxoxo** 
Are you the master gayageum player Han TeRra?? This is indeed a research opportunity! - I wish I knew enough about Magenta and neural nets to help you - please somebody work with this artist, this would be a phenomenal creative exchange!

**Comment by terraxoxo** [comment]
yes, ha! thank you, yes, work with me, explorer something, ! ha.

**Comment by Zenanimator** [comment]
:) Annyeonghaseyo. I hope someone steps up to help you for although I
recognize the tremendous value of a collaboration between the classical
Asian (Korean) musical arts and modern neural net research into sound
structures, I am almost completely ignorant of the technologies required to
do that research. I am studying cognitive science and philosophy (and am an
amateur musician) but I'm still years away from really being able to offer
any real help. I do truly wish you all the best and hope someone can help.
If I come across anyone in my research contacts looking to expand their
research in this direction I will refer them to you.

**Comment by terraxoxo** [comment]
oh, thank you very much zen animator. ha. do you like zen?
I am really interested in AI or other technologies also, virtual reality. I
am looking for someone who are interested in them, but i have no knowledge
even MIDI, i din't know that all these day's music are made by computer?
ha! so now am going to study it and make music with the midi. if anybody
know midi music even for pop music program, please let me know whatever.
best wishes, xoxo

**Comment by adarob** [comment]
Hi @terraxoxo, we are certainly interested in all types of music, but as other have pointed out, machine learning is a data-based science and we unfortunately currently lack sufficient amounts of non-western music in the data format we need. If someone were able to build a collection in either MIDI or MusicXML format, we would love to collaborate on building appropriate models for it! Thanks for reaching out, an please let us know if you'd like to discuss more.

**Comment by Zenanimator** [comment]
@terraxoxo I called up a friend of mine who's an experienced (now retired) recording engineer and musician about some concerns I was having about these recommendations to use MIDI. The thing about midi is is its going to trash all the subtlety in your music. Asian music (and here I'm including the sitar, which you say you also play) are absolutely reliant on microtonal variation... midi isn't going to capture any of that. I'm thinking someone might have to train a network to do some kind of Fourier pattern recognition on the waveform data itself...but I'm just guessing here. Melodyne does some of this and may be useful. Here's my friend's (more experienced) concurrence:

"I agree. MIDI is now a thirty-something year old data format that is not well suited to microtonal systems. 
"It seems that Celemony supports a wide range of alternate tunings, but does not include a specific Korean scale. It may allow the import of custom scale tunings. It does perform an esoteric FFT analysis on the audio to derive pitch/timing data for display but leaves the audio file untouched (until you edit…). I haven’t explored the alt-tuning capabillities in detail."

**Comment by g0dd4rd** [comment]
@Zenanimator i suggest you start with regular pentatonic but add little oscillation on its tones (i'd do a bend or a vibrato/tremolo on guitar to achieve that) to simulate microtones (intervals between tones that are smaller than a semitone). if i used a synth, i'd decide on a waveform (square, saw etc.) and then shifted the phase a bit. my opinion is that you can use midi as long as there's a fine grain resolution in order to achieve either one of the described methods.

**Comment by terraxoxo** [comment]
dear all, to be honest, I don't know how and where to start. MIDI studying
will be first? I only did until now ORGANIC instrument and vocal. so just
let me know real basic level to start computer music. and please let me
know the program which have to be ready.! thanks! xoxo

**Comment by DefinitlyEvil** [comment]
@terraxoxo understand midi, sheet music, harmony/counterpoint(for me those two are the same) and that will do for most of the western music... i guess

**Comment by terraxoxo** [comment]
I do all what you mentioned except for the 'midi'.
so please let me know how to start to study 'midi', is there any book or
program? or website?
I want to make pop music, is that with midi?

**Comment by Jishanshaikh4** [comment]
@terraxoxo You might want to consider exploring more about MIDI here at http://www.indiana.edu/~emusic/361/midi.htm I'm new to music generation and digitization. I'm more of programmer than a composer. Extremely interested in development of Asian music composition using magenta. I'm trying to convert standard melodies/songs to standard formats such as MusicXML. Hey @adarob, feel free to give more pointers on that.
